add binary-step-activation-function.md (#2542)
Codecademy/docs: content/ai/concepts/neural-networks/terms/binary-step-activation-function/binary-step-activation-function.md

---
Title: 'Binary Step Activation Function'

> Description: 'A binary-step function is the simplest threshold based activation function which works by either firing the node in case the threshold value is surpassed or doing nothing at all.'

Reviewer: ```suggestion
Description: 'The simplest threshold based activation function which works by either firing the node in case the threshold value is surpassed or doing nothing at all.'
```
Changed due to how Docs displays terms: Title : Description. Because of this, mentioning the title in a term's description is redundant.

Subjects:
  - 'Machine Learning'
  - 'Data Science'
  - 'Computer Science'
Tags:
  - 'AI'
  - 'Machine Learning'
  - 'Neural Networks'
  - 'Functions'
CatalogContent:
  - 'machine-learning'
  - 'path/data-science'
---

## Binary Step Activation Function

> **Binary Step** is the simplest threshold based activation function which works by either activating the node i.e., passing on its output to the next layer in case the threshold value is surpassed or doing nothing at all. 

Reviewer: ```suggestion
**Binary Step** is the simplest threshold-based activation function which works by either activating the node, i.e. passing on its output to the next layer in case the threshold value is surpassed, or doing nothing at all. 
```

![Binary Step Activation Plot](https://raw.githubusercontent.com/Codecademy/docs/main/media/binary-step-plot.png)

Mathematically, it can be defined as:

![Binary Step Activation Function](https://raw.githubusercontent.com/Codecademy/docs/main/media/binary-step-function.png)

If the node input value is less than or equal to 0, it returns 0 as output else 1.

### Usage and Limitation

As the name suggests, binary step can be used as an activation function in the output layer of a neural network for binary classification problems.

It is generally not suggested to use binary step in the hidden layer as it is a linear piecewise function unable to handle complex representations.

### Codebyte Example

> ```python

Reviewer: ````suggestion

The following is an example of the activation function in Python:

```codebyte/python
````

Reviewer: If we include this we want a sentence description of the example.

---
Title: 'Binary Step Activation Function'
Description: 'The simplest threshold based activation function which works by either firing the node in case the threshold value is surpassed or doing nothing at all.'
Subjects:
  - 'Machine Learning'
  - 'Data Science'
  - 'Computer Science'
Tags:
  - 'AI'
  - 'Machine Learning'
  - 'Neural Networks'
  - 'Functions'
CatalogContent:
  - 'machine-learning'
  - 'path/data-science'
---

## Binary Step Activation Function

**Binary Step** is the simplest threshold-based activation function which works by either activating the node, i.e. passing on its output to the next layer in case the threshold value is surpassed, or doing nothing at all. 

![Binary Step Activation Plot](https://raw.githubusercontent.com/Codecademy/docs/main/media/binary-step-plot.png)

Mathematically, it can be defined as:

![Binary Step Activation Function](https://raw.githubusercontent.com/Codecademy/docs/main/media/binary-step-function.png)

If the node input value is less than or equal to 0, it returns 0 as output else 1.

> ### Usage and Limitation

Reviewer 2: ```suggestion
### Usage and Limitations
```

As the name suggests, binary step can be used as an activation function in the output layer of a neural network for binary classification problems.

It is generally not suggested to use binary step in the hidden layer as it is a linear piecewise function unable to handle complex representations.

### Codebyte Example

The following is an example of the activation function in Python:

```codebyte/python
def binary_step(x):

>   """ Apply binary step activation"""

Reviewer 2: ```suggestion
  # Apply binary step activation
```

  if x <= 0:

>     # deactivate node

Reviewer 2: ```suggestion
    # Deactivate node
```

    return 0
  else:

>     # activate node

Reviewer 2: ```suggestion
    # Activate node
```

    return 1

> # examples

Reviewer 2: ```suggestion
# Examples
```
